### ISSUE TITLE
Package scope

### ISSUE THREAD

**Issue by gdkrmr** [gdkrmr/dimRed#60]
Regarding the downtime of `dimRed` on CRAN (see #57 and #58), I have done some thinking and while R packages are easy to maintain due to the great stability of the R language, `dimRed` is making the maintainer's life unnecessarily difficult by doing to many things.

The scope of the package needs to be better defined. `dimRed` is a wrapper around other packages and should not do more than that. This means that the following things we are currently doing are out of scope:

- Numeric accuracy testing
- Including own methods, e.g. Isomap and Autoencoder.

This means I will do the following:

- [ ] Remove tests that only test for numeric accuracy
- [ ] Keep tests for the APIs of the wrapped packages.
- [ ] Move methods into their own packages and wrap those.

Not sure yet what to do with complicated python dependencies (e.g. UMAP and the Autoencoder), I will probably push the burden onto the user and provide some helper functions, see e.g. `tensorflow` and `keras`.

I will probably keep documentation around, good documentation has always been a staple of R, there is also no real maintainer burden
